code/modules/chemistry/Chemistry-Tools.dm

```diff
@@ -1013,6 +1017,134 @@ proc/ui_describe_reagents(atom/A)
 			flick("dropper_funnel_swon",src)
 			icon_state = "dropper_funnel_on"
 
+
+
+/obj/item/reagent_containers/glass/plumbing/dispenser
+	name = "portable chemical dispenser"
+	desc = "A synthesizer, ripped from a Chem Dispenser, capable of placing reagents in glassware at a fixed rate."
+	icon = 'icons/obj/items/chemistry_glassware.dmi'
+	icon_state = "producer"
+	splash_all_contents = FALSE
+	object_flags = SUPPRESSATTACK
+	initial_volume = 0
+	accepts_lid = TRUE
+	rc_flags = FALSE
+	fluid_overlay_states = 0
+	container_style = "condenser"
+	var/running = FALSE
+	var/flow_rate = 1
+	var/reagent_to_fabricate = ""
+	var/fabrication_tick = 0
+	var/ticks_to_fabricate = 4
+	var/image/fabricating_overlay = null
+	var/image/fabrication_tick_lights = null
+	HELP_MESSAGE_OVERRIDE({"Click and drag this onto other glassware to connect it.
+	Flow rate and chemical fabricated can be adjusted either in your hand, or with a <b>screwdriver</b>.
+	To pick this up, click and drag it onto your person."})
+
+
+	get_desc()
+		. = " It is set to dispense [flow_rate] units of [reagent_to_fabricate]."
+
+	New()
+		. = ..()
+
+	disposing()
+		src.remove_container()
+		. = ..()
+
+	mouse_drop(atom/over_object, src_location, over_location)
+		if (ismob(over_object))
+			var/mob/user = over_object
+			if (user == usr && !user.restrained() && !user.stat && (user.contents.Find(src) || in_interact_range(src, user)))
+				if (!user.put_in_hand(src))
+					return ..()
+		else
+			..()
+
+	add_line(var/obj/container)
+		var/datum/lineResult/result = drawLine(src, container, "condenser", "condenser_end", src.pixel_x + 10, src.pixel_y, container.pixel_x, container.pixel_y + container.get_chemical_effect_position())
+		result.lineImage.pixel_x = -src.pixel_x
+		result.lineImage.pixel_y = -src.pixel_y
+		result.lineImage.layer = src.layer+0.01
+		src.UpdateOverlays(result.lineImage, "tube\ref[container]")
+	attack_self(mob/user as mob)
+		adjust_flow_rate(user)
+		return
+
+	attackby(obj/item/I, mob/user)
+		if (istype(I, /obj/item/screwdriver))
+			adjust_flow_rate(user)
+			user.visible_message(SPAN_NOTICE("[user.name] adjusts \the [src.name]."))
+			return
+		..()
+	process()
+		if (running)
+			fabrication_tick += 1
```

Comment:
Should ideally be mult adjusted, see `item.last_tick_duration` maybe?

Reply:
i'll leave it for now - see how slow it gets in practise. it jumping might mess up some weird complex flow i haven't thought of yet

Reply:
Fair actually, consistency might be better than speed.